### ISSUE TITLE
[Column Block] Improve labeling of nested columns

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#29804]
## What problem does this address?

Currently, columns are labeled very simply either as "Columns" for a parent block or "Column" for a nested block. When using a screen reader (NVDA specifically) this makes for a very confusing experience as noted in the following video because you're unable to know which nested column you're in:

https://user-images.githubusercontent.com/26996883/110888952-768dfc80-82aa-11eb-910a-1c80b385cc04.mov

I'm sharing this video with consent of the person who I recorded. 

## What is your proposed solution?

Add more descriptive labeling either using numbers or "right and left" indications to make it clear which column one is in when using a screen reader.

**Comment by carolinan** [comment]
This can probably be done by passing the count from the columns block to the inner column block using _[context](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/)_, but I was not able to solve it.
